### PR TITLE
Update rule sssd_enable_smartcards for OL8 and RHEL8

### PR DESCRIPTION
**Pull request by Xeicker**
#### Description:

- Update OVAL in rule `sssd_enable_smartcards` to check also `/etc/pam.d/system-auth` & `smartcard-auth`
- Update bash and ansible to remediate also `/etc/pam.d/system-auth` & `smartcard-auth` using `authselect` if available
- Add tests to verify this behavior

#### Rationale:

- This complies with `OL08-00-020250` and `RHEL-08-020250`

### CODE REVIEW

File: linux_os/guide/services/sssd/sssd_enable_smartcards/ansible/shared.yml [hunk 1]
```diff
@@ -33,3 +33,179 @@
...
+      when:
+        - result_pam_try_cert_auth_present_smartcard_auth.found == 0
+
+    - name: Ensure the try_cert_auth option is in system-auth
+      ansible.builtin.replace:
+        dest: "/etc/pam.d/system-auth"
+        regexp: '^(auth.*sufficient.*pam_sss\.so)((?!try_cert_auth).)*$'
+        replace: '\g<1> try_cert_auth \g<2>'
+      when:
+          - result_pam_try_cert_auth_present_system_auth.found == 0
```
Comment (jan-cerny) [inline linux_os/guide/services/sssd/sssd_enable_smartcards/ansible/shared.yml:207]: If I understand it well, the OVAL check requires that the setting is present in both smartcard-auth and system-auth, there is an AND between the criteria. So then the Ansible remdiation should create both of them.

File: linux_os/guide/services/sssd/sssd_enable_smartcards/ansible/shared.yml [hunk 2]
```diff
@@ -33,3 +33,179 @@
     value: 'true'
     create: yes
     mode: 0600
+
```
Comment (jan-cerny) [inline linux_os/guide/services/sssd/sssd_enable_smartcards/ansible/shared.yml:36]: I think that there is some problem in the Ansible remediation. Because the test scenarios pass with Bash remediation but fail with Ansible remediation.

```
[jcerny@thinkpad scap-security-guide{pr/8835}]$ python3 tests/test_suite.py rule --libvirt qemu:///system ssgts_rhel8 sssd_enable_smartcards 
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2022-05-27-0958/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_sssd_enable_smartcards
INFO - Script correct_value.pass.sh using profile (all) OK
INFO - Script pamd_argument_missing.fail.sh using profile (all) OK
INFO - Script pamd_argument_missing_authselect.fail.sh using profile (all) OK
INFO - Script value_missing.fail.sh using profile (all) OK
INFO - Script wrong_value.fail.sh using profile (all) OK
[jcerny@thinkpad scap-security-guide{pr/8835}]$ python3 tests/test_suite.py rule --remediate-using ansible --libvirt qemu:///system ssgts_rhel8 sssd_enable_smartcards 
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2022-05-27-1005/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_sssd_enable_smartcards
INFO - Script correct_value.pass.sh using profile (all) OK
INFO - Script pamd_argument_missing.fail.sh using profile (all) OK
ERROR - Rule evaluation resulted in fail, instead of expected pass during final stage 
ERROR - The check after remediation failed for rule 'xccdf_org.ssgproject.content_rule_sssd_enable_smartcards'.
INFO - Script pamd_argument_missing_authselect.fail.sh using profile (all) OK
ERROR - Rule evaluation resulted in fail, instead of expected pass during final stage 
ERROR - The check after remediation failed for rule 'xccdf_org.ssgproject.content_rule_sssd_enable_smartcards'.
INFO - Script value_missing.fail.sh using profile (all) OK
ERROR - Rule evaluation resulted in fail, instead of expected pass during final stage 
ERROR - The check after remediation failed for rule 'xccdf_org.ssgproject.content_rule_sssd_enable_smartcards'.
INFO - Script wrong_value.fail.sh using profile (all) OK
```

Have you experienced this? Can you take a look?

Comment (Xeicker) [reply linux_os/guide/services/sssd/sssd_enable_smartcards/ansible/shared.yml:36]: I didn't see it before, curious. Apparently ansible was missing an edge case when pam_sss.so is not set in auth

**Review by jan-cerny — APPROVED** [review]
Thanks!

**Comment by jan-cerny** [comment]
```
jcerny@thinkpad scap-security-guide{pr/8835}]$ python3 tests/test_suite.py rule --libvirt qemu:///system ssgts_rhel8 sssd_enable_smartcards
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2022-05-31-0952/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_sssd_enable_smartcards
INFO - Script correct_value.pass.sh using profile (all) OK
INFO - Script pamd_argument_missing.fail.sh using profile (all) OK
INFO - Script pamd_argument_missing_authselect.fail.sh using profile (all) OK
INFO - Script value_missing.fail.sh using profile (all) OK
INFO - Script wrong_value.fail.sh using profile (all) OK
[jcerny@thinkpad scap-security-guide{pr/8835}]$ python3 tests/test_suite.py rule --remediate-using ansible --libvirt qemu:///system ssgts_rhel8 sssd_enable_smartcards
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2022-05-31-0956/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_sssd_enable_smartcards
INFO - Script correct_value.pass.sh using profile (all) OK
INFO - Script pamd_argument_missing.fail.sh using profile (all) OK
INFO - Script pamd_argument_missing_authselect.fail.sh using profile (all) OK
INFO - Script value_missing.fail.sh using profile (all) OK
INFO - Script wrong_value.fail.sh using profile (all) OK
```
